### PR TITLE
Races with NOHUNGER won't vomit

### PR DESCRIPTION
**Pull request by XDTM**
They have nothing to throw up with.

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -440,43 +440,44 @@
...
-	if(stun)
-		Stun(4)
-
-	playsound(get_turf(src), 'sound/effects/splat.ogg', 50, 1)
-	var/turf/T = get_turf(src)
-	for(var/i=0 to distance)
-		if(blood)
-			if(T)
-				add_splatter_floor(T)
+	if(!(NOHUNGER in dna.species.specflags))
```
Comment (phil235) [inline code/modules/mob/living/carbon/carbon.dm]: aliens are carbons and do not have dna.

Comment (XDTM) [inline code/modules/mob/living/carbon/carbon.dm]: Added the dna check

**Comment by kevinz000** [comment]
could just do "if nohunger return" instead of putting the code into its block I think, might be better?

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -440,43 +440,44 @@
...
-	if(stun)
-		Stun(4)
-
-	playsound(get_turf(src), 'sound/effects/splat.ogg', 50, 1)
-	var/turf/T = get_turf(src)
-	for(var/i=0 to distance)
-		if(blood)
-			if(T)
-				add_splatter_floor(T)
+	if(dna && dna.species && !(NOHUNGER in dna.species.specflags))
```
Comment (ChangelingRain) [inline code/modules/mob/living/carbon/carbon.dm]: Do monkeys have species? If they don't, they won't be able to vomit.

**Comment by XDTM** [comment]
thanks kevinz000, i really should have thought of that; it should also prevent monkeys not vomiting

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -440,6 +440,9 @@
 	return ..()
 
 /mob/living/carbon/proc/vomit(var/lost_nutrition = 10, var/blood = 0, var/stun = 1, var/distance = 0, var/message = 1, var/toxic = 0)
+	if(dna && dna.species && !(NOHUNGER in dna.species.specflags))
+		return 1
```
Comment (CyberbossJHCB) [inline code/modules/mob/living/carbon/carbon.dm]: Forgive my ignorance if I'm wrong, won't this cause all things WITHOUT "NOHUNGER" to NOT throw up?

Comment (coiax) [inline code/modules/mob/living/carbon/carbon.dm]: return true

Comment (XDTM) [inline code/modules/mob/living/carbon/carbon.dm]: Oh shit you're right, i'll fix that as soon as i can

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**